### PR TITLE
Spec: Add missing error codes to REST API

### PR DESCRIPTION
**Pull request by rdblue**

### CODE REVIEW

File: rest_docs/rest-catalog-open-api.yaml [hunk 1]
```diff
@@ -192,6 +192,8 @@ paths:
           $ref: '#/components/responses/UnauthorizedResponse'
         403:
           $ref: '#/components/responses/ForbiddenResponse'
+        406:
+          $ref: '#/components/responses/UnsupportedOperationResponse'
```
Comment (rdblue) [inline rest_docs/rest-catalog-open-api.yaml:196]: This may happen if the service doesn't support custom metadata.

**Review by kbendick — APPROVED** [review]
This looks good to me.

**Review by nastra — APPROVED** [review]
LGTM, although the 406 http error code should rather be a 400 for `UnsupportedOperationResponse` imo as 406 more relates to the `Accept` header fields for content negotiation according to 
https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.6

**Review by RussellSpitzer — APPROVED** [review]
+1

**Comment by nastra** [comment]
> > 406 http error code should rather be a 400
> 
> I don't think that I agree. 400 is something that the caller did incorrectly, but this is that the service doesn't support custom metadata. Neither one fits very well, but I think 406 is closer because it indicates that something isn't supported.

yeah I guess it's difficult to find the appropriate error code for this particular one

**Comment by rdblue** [comment]
Thanks for the reviews, @kbendick, @nastra, and @RussellSpitzer!

**Pull request merged**